doc/contributing/coding-conventions.chapter.md

```diff
@@ -196,7 +196,23 @@
   As an exception, an explicit conditional expression with null can be used when fixing a important bug without triggering a mass rebuild.
   If this is done a follow up pull request _should_ be created to change the code to `lib.optional(s)`.
 
-- Arguments should be listed in the order they are used, with the exception of `lib`, which always goes first.
+- The order of elements in the argument set must follow the convention below:
+
+  01. `lib`;
+  02. `stdenv`;
+  03. builders (`buildPythonPackage`, `mkDerivation`, `buildGoModule`);
+  04. fetchers (`fetchurl`, `fetchpatch`, `fetchFromGitHub`);
+  05. `buildPackages`;
+  06. all the remaining arguments (i.e. those that don't fit the other items);
+  07. options (`extraModules`, `disabledProtocols ? [ "tcp"  "udp" ]`);
+  08. Boolean flags (`enableSDL`);
+  09. testers (`nixosTests`, `testers`);
+  10. updaters (`gitUpdater`, `gitUnstableUpdater`);
+  11. Darwin-related arguments (`darwin`, `Metal`, `IOKit`).
```

Comment:
I don't think darwin should be special cased here, this can be merged into `all the remaining arguments`

Reply:
This is more because of tradition. They always appeared last, sometimes with a comment like `# Darwin dependencies`.

I believe this is good, because they are dealt only when some specific Darwin quirk comes into the picture.

Keeping Darwin dependencies in a dedicated block helps when someone proposes a Darwin-specific change.